[SM-515] Add reference to the use of secrets (#47)
mulesoft/docs-service-mesh: modules/ROOT/pages/create-an-api-configure-service-mesh-CRD.adoc

Anypoint Service Mesh supports the use of Kubernetes secrets to store and manage sensitive information, such as your user credentials.

> *Task Prerequisites*

Reviewer: The thing I don't like about using this as an inline heading is that it is a heading for only one item and, more importantly, there is no indication that the actual task begins next.
Are the steps that follow how you MEET the prerequisites?
It's unclear.

Anypoint Service Mesh supports the use of Kubernetes secrets to store and manage sensitive information, such as your user credentials.

*Task Prerequisites*

> You must have the Devops or Admin role or permission to create the Kubernetes secret.

Author: ```suggestion
You must have permission to create the Kubernetes secret.
```

Reviewer 2: Done! Thanks!

> You must have the Devops or Admin role or permission to create the Kubernetes secret.

Reviewer: Is "Devops" actually in the UI? I've never seen this as a role name.

Author: Sorry, maybe we have some confusion around that, we haven't any roles in the platform, we need to change it to `You must have permission to create the Kubernetes secret.`

Reviewer 2: Done! thanks!

> //[PA: Rodrigo--Are these the exact nams of the roles or permissions?]

Author: ```suggestion
```

Reviewer 2: Removed!

To store your user credentials in Kubernetes secrets:

> . In a `.yaml` file, create a secret in which to store the sensitive information, for example `orders-api-config.yaml`:

Reviewer: See the style reference entry for file types vs. file extensions:
```suggestion
. In a a YAML file, create a secret in which to store the sensitive information, for example, `orders-api-config.yaml`:
```

Reviewer 2: Done! Thanks!

+
[source,text,linenums]
----
apiVersion: v1
kind: Secret
metadata:
  name: <secret name> # example: orders-api-config
  namespace: <namespace name> # example: default
type: Opaque
stringData:
  config: |
    {
      "user": "<replace-with-anypoint-username>",
      "password": "<replace-with-anypoint-clientSecret>"
    }
----
. Apply the secret: 
+
`kubectl apply -f orders-api-config.yaml`
+

> If the secret is correctly applied, a "secret/orders-api-config created" message is displayed.

Reviewer: ```suggestion
If the secret is correctly applied, a 'secret/orders-api-config created' message is displayed.
```

Reviewer: See style guide entries for font use.

Reviewer 2: Done! Thanks!

+

> . Reference the secret in your API CRD, (instead of the user and password parameter), as illustrated in the following example:

Reviewer: ```suggestion
. Reference the secret in your API CRD (instead of the user and password parameter), as illustrated in the following example:
```

Reviewer 2: Done! Thanks!

+
[source,text,linenums]
----
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceInstance
metadata:
  name: <api name> # example: orders-api
  namespace: <namespace name> # example: default
  labels:
    service-mesh.mulesoft.com/kind: api
spec:
  clusterServiceClassExternalName: anypoint-platform-api-instance
  clusterServicePlanExternalName: service-mesh
  parameters:
    environmentId: <environment ID> # example: 7cea45c6-3237-464d-bd34-3122c4abde29
    groupId: <group ID> # example: 350b8164-e809-4091-91f5-079fb30e5a1e
    assetId: <asset ID> # example: orders-api
    version: <version> # example: 0.0.1
    instanceLabel: <instance label> # example: public
    tags:
      apiSpec:
      - <tag in Exchange asset> # example: orders
      apiInstance:
      - <tag in API Manager instance> # example: orders

>   parametersFrom: # we will refer to the secret holding the config from here

Reviewer: "the secret holding the config from here"?
I know this is just an internal comment, but it's pretty unclear, at least to me....

Author: We need to find some way to refer to the previous created secrets, feedback is welcome

Reviewer 2: The element specifies the secret name and configuration required for retrieving the parameters.
